src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +31,70 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfig(remoteCert *x509.Certificate) (*tls.Config, error) {
+	if remoteCert == nil {
+		return nil, fmt.Errorf("Invalid remote public key")
```

Comment:
errors are always lower-case. The handler of the error will make this a sentence at the end.

```suggestion
		return nil, fmt.Errorf("invalid remote public key")
```